metrics/source_stats_reporter.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metrics
+
+import (
+	"context"
+	"strconv"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics/metricskey"
+)
+
+var (
+	// eventCountM is a counter which records the number of events sent by the source.
+	eventCountM = stats.Int64(
+		"event_count",
+		"Number of events sent",
+		stats.UnitDimensionless,
+	)
+)
+
+type ReportArgs struct {
+	Namespace     string
+	EventType     string
+	EventSource   string
+	Name          string
+	ResourceGroup string
+}
+
+// StatsReporter defines the interface for sending source metrics.
+type StatsReporter interface {
+	ReportEventCount(args *ReportArgs, responseCode int) error
```

Comment:
Could you add a comment for this func? Looks like it only reports one per call?

Reply:
Yes, added a comment. Thanks!